packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -34,21 +46,67 @@ public function __construct(string $definitionDirectory, string $indexPrefix, Cl
         $this->client = $client;
     }
 
+    /**
+     * @param int $buildVersion
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setBuildVersion(int $buildVersion): void
+    {
+        if ($this->buildVersion !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->buildVersion = $buildVersion;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Setting\Setting $setting
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setSetting(Setting $setting): void
+    {
+        if ($this->setting !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->setting = $setting;
+    }
+
     /**
      * @param int $domainId
      * @param string $index
      * @return string
      */
     public function getIndexName(int $domainId, string $index): string
     {
-        return $this->indexPrefix . $index . $domainId;
+        return $this->indexPrefix . $this->buildVersion . $index . $domainId;
+    }
+
+    /**
+     * @param int $domainId
+     * @return string
+     */
+    public function getCurrentProductIndexName(int $domainId)
```

Comment:
There are couple of methods to get an index name:
* `getIndexName`
* `getAliasName`
* `getCurrentProductIndexName`

When should I use what method? The naming doesn't help me unfortunately

Reply:
This is related with the new terminology: index name, alias name, current product index name, alias for current index, current alias, current index.
The terminology must be simple and consistent to be understandable. I cannot give you suggestion because I don't understand what methods even do